02-peer-protocol.md

```diff
@@ -1713,6 +1762,161 @@ satoshis, which is possible if `dust_limit_satoshis` is below 546 satoshis).
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - otherwise:
+    - MUST sign and broadcast the corresponding closing transaction.
+
+### Rationale
+
+The close protocol is designed to avoid any failure scenarios caused by fee disagreement,
+since each side offers to pay its own desired fee.
+
+If one side has less funds than the other, it may choose to omit its own output, and in this case
+dust MUST be omitted, to ensure that the resulting transaction can be broadcast.
+
+The corner case where fees are so high that both outputs are dust is addressed in two ways: paying
+a low fee to avoid the problem, or using an `OP_RETURN` (which is never "dust"). If one side
+chooses to use an `OP_RETURN` output, its amount must be 0 to ensure that the resulting transaction
+can be broadcast.
+
+Note that there is usually no reason to pay a high fee for rapid processing, since an urgent child
+could pay the fee on the closing transactions' behalf. If rapid processing is desired and CPFP is
+not an option, the closer can RBF its previous closing transactions by sending `shutdown` again.
+
+Sending a new `shutdown` message overrides previous ones, so you can negotiate again (even changing
+the output address when `upfront_shutdown_script` was not negotiated) if you want: in this case
+there's a race where you could receive `closing_complete` for the previous output address, and the
+signature won't validate. In this case, ignoring the `closing_complete` is the correct behaviour,
+as the new `shutdown` will trigger a new `closing_complete` with the correct signature. This
+assumption that we only remember the last-sent of any message is why so many cases of bad
+signatures are simply ignored.
+
+When sending a new `shutdown`, we must receive a new `shutdown` from the remote node before
+sending `closing_complete`. This is necessary to be compatible with future taproot channels
+that use musig2 and need to exchange random nonces every time a transaction spending the channel
+output is signed. Note that the remote `shutdown` doesn't need to be an explicit response to the
+local `shutdown` that was sent: if both nodes concurrently send `shutdown`, they can exchange
+`closing_complete` immediately after receiving the remote `shutdown`.
+
+If the closer proposes a transaction which will not relay (an output is dust, or the fee rate it
+proposes is too low), it doesn't harm the closee to sign the transaction.
+
+Similarly, if the closer proposes a high fee, it doesn't harm the closee to sign the transaction,
+as the closer is paying.
+
+There's a slight game where each side would prefer the other side pay the fee, and proposes a
+minimal fee. If neither side proposes a fee which will relay, the negotiation can occur again,
+or the final commitment transaction can be spent. In practice, the opener has an incentive to
```

Comment:
Nit:

When creating the mutual close, the final commitment transaction hasn't been broadcast yet, so it cannot be spent yet right?

I suggest something like: "or the final commitment transaction can be broadcast and spent."

Or is that what is meant by "to fail the channel" elsewhere in the spec? Thank you.

Reply:
Yeah, it means force close by signing and broadcasting commitment transaction in this case.

The explanation is in the introduction (bolt 0):
https://github.com/lightning/bolts/blob/aa5207aeaa32d841353dd2df3ce725a4046d528d/00-introduction.md?plain=1#L135-L141